application/models/TokenDynamic.php

```diff
@@ -1105,4 +1105,30 @@ public function getSurveyId()
     {
         return self::$sid;
     }
+
+    /**
+     * Returns true if the token can be used
+     * @return bool
+     */
+    public function canBeUsed()
+    {
+        return !empty($this->token)
+            && (
+                $this->completed == "N"
+                || empty($this->completed)
```

Comment:
No `$this->usesleft > 0 `check ? (same than email)

Reply:
That was ported from the gridButtons part.
I think that could be part of `canBeUsed` but have some doubts about scenarios, ex: editing a prior response.
Anyway, don't want to update anything now, jusr reog a bit the code to start cetralizing business rules.